File: macos/.config/skhd/launcher/util_func.lua
local util_func = {}

function util_func.has_key (tab, val)
    for index, value in ipairs(tab) do
        if value.key == val then
            return index
        end
    end
    return false
end

function util_func.readf(cmd)
    local fileHandle = io.popen(cmd)
    local output = fileHandle:read('*a')
    fileHandle:close()
    return output
end

function util_func.readkey()
    local readkey_f = os.getenv('HOME') .. '/.local/bin/readkey.sh'
    return util_func.readf(readkey_f):sub(1, 1)
end

return util_func

File: macos/.config/skhd/launcher/other/c/util_func.c
#include "util_func.h"

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

char gkey()
{

    FILE *fp;
    char key;

    /* Open the command for reading. */
    fp = popen("bash -c '~/.local/bin/readkey.sh'", "r");
    if (fp == NULL) {
        printf("Failed to run command\n");
        exit(1);
    }

    /* Read the output a line at a time - output it. */
    key = fgetc(fp);

    /* close */
    pclose(fp);

    return key;
}

int key_apps(char key, struct App apps[], int n)
{
    for (int i = 0; i < n; i++) {
        if (apps[i].key == key)
            return i;
    }
    return -1;
}

// int main()
// {
//     char key;
//     key = gkey("ls -l");
//     printf("%c\n", key);
//     return 0;
// }
